dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java

```diff
@@ -858,79 +873,96 @@ private void setHTMLPagesDependencies(final Set<String> idsToWork,final Publishe
 	private void setTemplateDependencies(final PublisherFilter publisherFilter)  {
 		try {
 			final List<Container> containerList = new ArrayList<>();
-			final FolderAPI folderAPI = APILocator.getFolderAPI();
 
 			for (final String id : templatesSet) {
-				final Template workingTemplate = APILocator.getTemplateAPI().findWorkingTemplate(id, user, false);
-				final Template liveTemplate = APILocator.getTemplateAPI().findLiveTemplate(id, user, false);
+				setTemplateDependencies(publisherFilter, containerList, id);
+			}
 
-				// Host dependency
-				if (!publisherFilter.doesExcludeDependencyClassesContainsType(PusheableAsset.SITE.getType())) {
-					final Host host = APILocator.getHostAPI()
-							.find(APILocator.getTemplateAPI().getTemplateHost(workingTemplate).getIdentifier(),
-									user, false);
-					hosts.addOrClean(
-							APILocator.getTemplateAPI().getTemplateHost(workingTemplate).getIdentifier(),
-							host.getModDate());
-				}
+		} catch (DotSecurityException e) {
 
-				containerList.clear();
-				// Container dependencies
-				if (!publisherFilter.doesExcludeDependencyClassesContainsType(PusheableAsset.CONTAINER.getType())) {
-					containerList.addAll(APILocator.getTemplateAPI()
-							.getContainersInTemplate(workingTemplate, user, false));
+			Logger.error(this, e.getMessage(),e);
+		} catch (DotDataException e) {
```

Comment:
The DotSecurityException and the DotDataException can be wrapped in the same catch